base/show.jl

```diff
@@ -20,41 +20,32 @@ struct IOContext{IO_t <: IO} <: AbstractPipe
     end
 end
 
-"""
-    IOContext(io::IO; properties...)
+unwrapcontext(io::IO) = io, ImmutableDict{Symbol,Any}()
+unwrapcontext(io::IOContext) = io.io, io.dict
```

Comment:
I introduced these two methods as they often allow to merge two methods (for `IO` and for `IOContext`) into one; moreover, they make it easy to spot what is discarded (e.g. in `IOContext(unwrapcontext(io)[1], unwrapcontext(context)[2])`, only the stream of `io` and the dict of `context` are kept).